x-pack/metricbeat/module/openai/usage/usage.go

```diff
@@ -172,12 +153,16 @@ func (m *MetricSet) fetchSingleDay(dateStr, apiKey string, httpClient *RLHTTPCli
 
 	resp, err := httpClient.Do(req)
 	if err != nil {
+		if err, ok := err.(net.Error); ok && err.Timeout() {
+			return ErrHTTPClientTimeout
+		}
 		return fmt.Errorf("error making request: %w", err)
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("error response from API: %s", resp.Status)
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("error response from API: status=%s, body=%s", resp.Status, string(body))
```

Comment:
be careful here to do not leak sensitive information. I usually try to log the body in its own log, without including the body in the error message

Reply:
I was trying this morning only. I did found one case where the logging the body is not ideal. In case of invalid API keys, the body shows the API key that it's invalid to connect to OpenAI. So, in case there's a typo when typing the API key, most part of the API key is exposed. So, I'll remove this in order to avoid any sensitive info leakage.